apps/retrospective/services/sprint.go

```diff
@@ -24,6 +24,25 @@ type SprintService struct {
 	DB *gorm.DB
 }
 
+// Check if a given sprint is deletable or not
+func (service SprintService) isSprintDeletable(sprint retroModels.Sprint) (bool, error) {
+	db := service.DB
+	// Check if the sprint is an active sprint sandwiched b/w a draft and a frozen sprint.
+	if sprint.Status == retroModels.ActiveSprint {
+		type SprintCount struct {
+			DraftCount, FreezeCount int
+		}
+		var sprintCount SprintCount
+		err := db.Model(&retroModels.Sprint{}).
```

Comment:
This query doesn't handle Discarded sprints and also the sprint count should be localized to the respective retrospective.
- Add a check for discarded sprints.
- Add a check for retrospective.